.github/workflows/main.yml

```diff
@@ -0,0 +1,134 @@
+name: FAIRDataPoint-index CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  btp:
+    name: Build-Tag-Push
+    runs-on: ubuntu-latest
+
+    env:
+      PUBLIC_IMAGE: fairdata/fairdatapoint-index
+      PRIVATE_IMAGE: ${{ secrets.PRIVATE_REGISTRY_URL }}/fairdatapoint-index
+      TAG_DEVELOP: develop
+      TAG_LATEST: latest
+      JDK_VERSION: 11
+      JDK_FILE: openjdk-11.0.2_linux-x64_bin.tar.gz
+      JDK_URL: https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_linux-x64_bin.tar.gz
+
+    services:
+      mongo:
+        image: mongo:4.2
+        ports:
+          - 27017:27017
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Prepare JDK folder
+        run: mkdir -p ~/jdk
+
+      # (1) -> Prepare cache and Java
+      - name: Cache ~/.m2
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+
+      - name: Cache JDK folder
+        uses: actions/cache@v1
+        with:
+          path: ~/jdk
+          key: ${{ env.JDK_FILE }}
+
+      # (2) -> Prepare Java
+      - name: Download Oracle JDK
+        run: |
+          if [ ! -f ~/jdk/$JDK_FILE ]; then
+            wget --quiet $JDK_URL -O ~/jdk/$JDK_FILE
+          fi
+          cp ~/jdk/$JDK_FILE .
+
+      - name: Setup Java
+        uses: actions/setup-java@master
+        with:
+          version: ${{ env.JDK_VERSION }}
+          jdkFile: ${{ env.JDK_FILE }}
```

Comment:
Honestly, I copied the one that I (painfully) did for FDP. The difference is in that here is Oracle JDK whereas they use Zulu JDK by default. This was causing problems with requests in tests in FDP but maybe it would work here... Should I try to simplify this here or should we stick with Oracle?

Reply:
Seems to be working without any issue... so let's use OpenJDK instead of Oracle 😊

Reply:
I got the impression they are defaulting to an OpenJDK distribution (due to licensing with the Oracle version). Let's keep an eye on it.